include/deal.II/base/point.h

```diff
@@ -419,7 +443,7 @@ Point<dim,Number>::square () const
 {
   Number q = Number();
   for (unsigned int i=0; i<dim; ++i)
-    q += this->values[i] * this->values[i];
+    q += numbers::NumberTraits<Number>::abs_square(this->values[i]);
```

Comment:
That's a change of behavior for complex data types. Can you add a note to the documentation of the function how it treats different `Number` types? In other words, how it interprets the term `square`.